src/formatDate.js

```diff
@@ -48,9 +48,57 @@
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dictionary = {};
+  const dateValues = date.split(fromFormat[3]);
+
+  // destructuring fromFormat into a dictionary
+  // for reading the parameter date
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'DD') {
+      dictionary['DD'] = dateValues[i];
+    } else if (fromFormat[i] === 'MM') {
+      dictionary['MM'] = dateValues[i];
+    } else if (fromFormat[i] === 'YY') {
+      dictionary['YY'] = dateValues[i];
+    } else if (fromFormat[i] === 'YYYY') {
+      dictionary['YYYY'] = dateValues[i];
+    }
+  }
+
+  let result = '';
+
+  // applying dictionary with toFormat
+  // to create the result (resulting date)
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'DD') {
+      result += dictionary['DD'];
+    } else if (toFormat[i] === 'MM') {
+      result += dictionary['MM'];
+    } else if (toFormat[i] === 'YY') {
+      if ('YY' in dictionary) {
+        result += dictionary['YY'];
+      } else {
+        result += dictionary['YYYY'].substring(2);
+      }
+    } else if (toFormat[i] === 'YYYY') {
+      if ('YYYY' in dictionary) {
+        result += dictionary['YYYY'];
+      } else {
+        if (dictionary['YY'][0] < '3') {
+          result += '20' + dictionary['YY'];
+        } else {
+          result += '19' + dictionary['YY'];
+        }
```

Comment:
Too many `if/else`!
Please simplify
I think the whole solution can ve written in 10 lines